workspaces/global-floating-action-button/plugins/global-floating-action-button/src/components/FAB.tsx

```diff
@@ -16,49 +16,104 @@
 import Typography from '@mui/material/Typography';
+import OpenInNewIcon from '@mui/icons-material/OpenInNew';
 import { FabIcon } from './FabIcon';
 import { FloatingActionButton, Slot } from '../types';
 
+const useStyles = makeStyles(() => ({
+  openInNew: { paddingBottom: '5px', paddingTop: '3px' },
+}));
+
 export const FAB = ({
   actionButton,
   size,
+  className,
 }: {
   actionButton: FloatingActionButton;
   size?: 'small' | 'medium' | 'large';
+  className?: string;
 }) => {
   const navigate = useNavigate();
+  const styles = useStyles();
   const isExternalUri = (uri: string) => /^([a-z+.-]+):/.test(uri);
-  const external = isExternalUri(actionButton.to!);
-  const newWindow = external && !!/^https?:/.exec(actionButton.to!);
+  const isExternal = isExternalUri(actionButton.to!);
+  const newWindow = isExternal && !!/^https?:/.exec(actionButton.to!);
   const navigateTo = () =>
-    actionButton.to && !external ? navigate(actionButton.to) : '';
+    actionButton.to && !isExternal ? navigate(actionButton.to) : '';
+  const labelText =
+    actionButton.label.length > 20
+      ? `${actionButton.label.slice(0, actionButton.label.length)}...`
+      : actionButton.label;
+  const getColor = () => {
+    if (actionButton.color) {
+      return actionButton.color;
+    }
+    if (!className) {
+      return 'info';
+    }
+    return undefined;
+  };
+
   return (
     <Tooltip
       title={actionButton.toolTip}
-      placement={Slot.PAGE_END ? 'left' : 'right'}
+      placement={actionButton.slot === Slot.PAGE_END ? 'left' : 'right'}
     >
-      <div>
+      <div className={className}>
         <Fab
           {...(newWindow ? { target: '_blank', rel: 'noopener' } : {})}
+          style={{ color: 'var(--pf-t--global--icon--color--regular)' }}
           variant={
-            actionButton.showLabel || !actionButton.icon
-              ? 'extended'
-              : 'circular'
+            actionButton.showLabel || isExternal ? 'extended' : 'circular'
           }
           size={size || actionButton.size || 'medium'}
-          color={actionButton.color || 'default'}
+          color={getColor()}
           aria-label={actionButton.label}
+          data-testid={actionButton.label
+            .replace(' ', '-')
+            .toLocaleLowerCase('en-US')}
           onClick={actionButton.onClick || navigateTo}
-          {...(external ? { href: actionButton.to } : {})}
+          {...(isExternal ? { href: actionButton.to } : {})}
         >
-          {actionButton.icon && <FabIcon icon={actionButton.icon} />}
-          {(actionButton.showLabel || !actionButton.icon) && (
-            <Typography sx={actionButton.icon ? { ml: 1 } : {}}>
-              {actionButton.label}
-            </Typography>
+          {(actionButton.slot === Slot.PAGE_END || !actionButton.slot) && (
+            <>
+              {isExternal && (
+                <OpenInNewIcon className={styles.openInNew} sx={{ mr: 1 }} />
+              )}
```

Comment:
Why couldn't we render this as before and just apply different CSS to change the order of the child elements?